roles/openshift_grafana/tasks/install_grafana.yaml

```diff
@@ -227,8 +231,8 @@
 - name: Add node exporter dashboard
   uri:
     url: "{{ grafana_route }}/api/dashboards/db"
-    user: admin
-    password: admin
+    user: "{{ grafana_sa_token.stdout }}"
+    force_basic_auth: true
```

Comment:
Yes, I did test and verify that the datasource/dashboard submissions via the API work just fine.

By using the grafana SA's token you authenticate as that SA, i.e. the request is authenticated as the grafana service account that the role creates instead of user "admin" (which we can't assume it exists).

The way grafana and oauth-proxy are currently configured, the grafana container [just grants admin privileges to unauthenticated (anonymous) requests](https://github.com/openshift/openshift-ansible/blob/openshift-ansible-3.10.0-0.4.0/roles/openshift_grafana/templates/grafana-config.yml.j2#L167-L174), therefore delegating all authentication to the oauth-proxy container, which only checks if the authenticated user has [permissions to list services in the grafana namespace](https://github.com/openshift/openshift-ansible/blob/openshift-ansible-3.10.0-0.4.0/roles/openshift_grafana/templates/grafana.yml.j2#L57). The grafana SA does have that permission to list services in its namespace.

I didn't set that configuration, I'm just using it here to avoid hardcoded credentials that are unlikely to apply to clusters in general... There might be room for improvement in this configuration, but I believe that would be a bit out of scope for this PR, whose main goal is to get the role to actually work in general with the current configuration options.

Reply:
Cool :) Thanks :)